examples/decoding/parallel_decoding.py

```diff
@@ -0,0 +1,245 @@
+print(f"Total frames to decode: {len(all_indices)}")
+
+
+# %%
+# Method 1: Sequential decoding (baseline)
+# ----------------------------------------
+#
+# Let's start with a sequential approach as our baseline. This processes
+# frames one by one without any parallelization.
+
+def decode_sequentially(indices: List[int], video_path=long_video_path):
+    """Decode frames sequentially using a single decoder instance."""
+    decoder = VideoDecoder(video_path, seek_mode="approximate")
+    return decoder.get_frames_at(indices)
+
+
+times, result_sequential = bench(decode_sequentially, all_indices)
+sequential_time = report_stats(times, unit="s")
+
+
+# %%
+# Method 2: FFmpeg-based parallelism
+# ----------------------------------
+#
+# FFmpeg has built-in multithreading capabilities that can be controlled
+# via the ``num_ffmpeg_threads`` parameter. This approach uses multiple
+# threads within FFmpeg itself to accelerate decoding operations.
+
+def decode_with_ffmpeg_parallelism(indices: List[int], num_threads: int, video_path=long_video_path):
+    """Decode frames using FFmpeg's internal threading."""
+    decoder = VideoDecoder(video_path, num_ffmpeg_threads=num_threads, seek_mode="approximate")
+    return decoder.get_frames_at(indices)
+
+
+NUM_CPUS = cpu_count()
+
+times, result_ffmpeg = bench(decode_with_ffmpeg_parallelism, all_indices, num_threads=NUM_CPUS)
+ffmpeg_time = report_stats(times, unit="s")
+speedup = sequential_time / ffmpeg_time
+print(f"Speedup vs sequential: {speedup:.2f}x with {NUM_CPUS} FFmpeg threads.")
+
+
+# %%
+# Method 3: multiprocessing
+# -------------------------
+#
+# Process-based parallelism distributes work across multiple Python processes.
+
+def decode_with_multiprocessing(indices: List[int], num_processes: int, video_path=long_video_path):
+    """Decode frames using multiple processes with joblib."""
+    chunks = split_indices(indices, num_chunks=num_processes)
+
+    results = Parallel(n_jobs=num_processes, backend="loky", verbose=0)(
+        delayed(decode_sequentially)(chunk, video_path) for chunk in chunks
+    )
+
+    return torch.cat([frame_batch.data for frame_batch in results], dim=0)
+
+
+times, result_multiprocessing = bench(decode_with_multiprocessing, all_indices, num_processes=NUM_CPUS)
+multiprocessing_time = report_stats(times, unit="s")
+speedup = sequential_time / multiprocessing_time
+print(f"Speedup vs sequential: {speedup:.2f}x with {NUM_CPUS} processes.")
+
+
+# %%
+# Method 4: Joblib multithreading
+# -------------------------------
+#
+# Thread-based parallelism uses multiple threads within a single process.
+# TorchCodec releases the GIL, so this can be very effective.
+
+def decode_with_multithreading(indices: List[int], num_threads: int, video_path=long_video_path):
+    """Decode frames using multiple threads with joblib."""
+    chunks = split_indices(indices, num_chunks=num_threads)
+
+    results = Parallel(n_jobs=num_threads, prefer="threads", verbose=0)(
```

Comment:
Does "prefer" mean that in some situations it might end up using processes? Does it default to processes, since we didn't say this above?